File: populate_db.py
import asyncio
from sqlalchemy.ext.asyncio import AsyncSession
from main import engine, SessionLocal, Item

async def insert_sample_data():
    async with SessionLocal() as session:
        async with session.begin():
            session.add_all([
                Item(name="Apple"),
                Item(name="Banana"),
                Item(name="Cherry"),
                Item(name="Date"),
                Item(name="Elderberry"),
                Item(name="Fig"),
                Item(name="Grape"),
                Item(name="Honeydew")
            ])
        await session.commit()

# Run the function to populate the database
asyncio.run(insert_sample_data())

File: populate_db.js
const { Sequelize, DataTypes } = require('sequelize');

// Initialize SQLite database
const sequelize = new Sequelize({
  dialect: 'sqlite',
  storage: './search.db',
});

// Define the model
const Item = sequelize.define('Item', {
  name: {
    type: DataTypes.STRING,
    allowNull: false,
  },
});

// Function to populate database
async function insertSampleData() {
  await sequelize.sync({ force: true }); // Reset database
  await Item.bulkCreate([
    { name: "Apple" },
    { name: "Banana" },
    { name: "Cherry" },
    { name: "Date" },
    { name: "Elderberry" },
    { name: "Fig" },
    { name: "Grape" },
    { name: "Honeydew" }
  ]);
  console.log("Database populated successfully!");
}

// Run the function
insertSampleData().then(() => {
  sequelize.close();
}).catch(err => console.error(err));
